validphys2/src/validphys/core.py

```diff
@@ -534,28 +537,49 @@ def __str__(self):
         return self.name
 
 class FKTableSpec(TupleComp):
-    def __init__(self, fkpath, cfactors):
+    """
+    In Legacy Mode each fktable has one single path.
+    For pineappl tables instead a FKTable is formed by any number of grids
+    therefore in order to check whether we have a new-format or old-format table
+    we will just check whether fkpath is a list
+    For now holds the metadata as an attribute to this function.
+    This is useless/transitional since this metadata is already in the new CommonData format
+    """
+    def __init__(self, fkpath, cfactors, metadata=None, norm=False):
+        self.cfactors = cfactors if cfactors is not None else []
+        self.legacy = True
+        if isinstance(fkpath, (tuple, list)):
+            self.legacy = False
+            fkpath = tuple(fkpath)
         self.fkpath = fkpath
-        self.cfactors = cfactors
+        self.metadata = metadata
+        self.norm = norm
         super().__init__(fkpath, cfactors)
 
-    #NOTE: We cannot do this because Fkset owns the fktable, and trying
-    #to reuse the loaded one fails after it gets deleted.
-    #@functools.lru_cache()
-    def load(self):
+    def _load_legacy(self):
```

Comment:
Can we not just delete this?

Reply:
No, I tried but it is needed in several places https://github.com/NNPDF/nnpdf/blob/2ae49281c5d471be490ed5c7bbd28e9211091c19/validphys2/src/validphys/dataplots.py#L792

I was "hunting them down" but at a certain point there was closure stuff involved so undid that. I might try again next week, it might have been something easy but I got a bit overwhelmed.

Reply:
Right. Better to not add more stuff here indeed.

Reply:
By here you mean this PR? Ok, then I'll add this to my todo list for the (near) future. As said in the code meeting, during the fit it doesn't seem to be used at all so it should be possible to remove it completely.